src/renderer/appIde/services/ProjectService.ts

```diff
@@ -358,12 +344,7 @@ class ProjectService implements IProjectService {
   }
 
   performAllDelayedSavesNow(): Promise<void> {
-    return this._asYouTypeSavesCount > 0 ?
-      new Promise<void>((resolve, _) => {
-        this._onDelayedSavesComplete.on(resolve);
-        this.runAllDelayedActionsInstantly();
-      })
-      : Promise.resolve();
+    return this._delayedJobs.instantlyRunAllOf(JOB_KIND_SAVE_FILE);
```

Comment:
This part I love the most. Here we don't have that ugly counter and event subscription anymore. Thanks the DelayedJobs entity!